src/main/java/org/terasology/taskSystem/TaskManagementSystem.java

```diff
@@ -344,4 +349,56 @@ public boolean assignAdvancedTaskToOreon(Actor oreon, String assignedTaskType) {
 
             return true;
     }
+
+    /**
+     * Receives the HorizontalCollisionEvent and decides whether the Oreon should abandon tasks
+     */
+    @ReceiveEvent(priority = EventPriority.PRIORITY_CRITICAL)
+    public void receiveCollisionEvent(HorizontalCollisionEvent collisionEvent, EntityRef oreon, MinionMoveComponent moveComponent) {
+        if (lastCollisionLocation == null) {
+            lastCollisionLocation = collisionEvent.getLocation();
+        } else {
+            if (lastCollisionLocation.getX() == collisionEvent.getLocation().getX() && lastCollisionLocation.getZ() == collisionEvent.getLocation().getZ()) {
+                logger.info("oreon stuck");
+                moveComponent.target = null;
+                oreon.saveComponent(moveComponent);
+                abandonTask(oreon);
+            } else {
+                //if collision just took place once
```

Comment:
Wouldn't this mean `// if collision did not take place`?
And do you need to update lastCollisionLocation to the current collision location if the value changes instead of changing it to null?   This is a guess, not a statement of fact....

Reply:
This method is called every time the Oreon collides with a block. So, if the Oreon was able to find a way around the last block it collided with, the `lastCollisionLocation` is set to `null` so that it would be updated with the new block's location upon next collision.